app/src/fdroid/java/org/fossasia/openevent/general/settings/SettingsFragment.kt

```diff
@@ -0,0 +1,110 @@
+
+        val activity =  activity as? AppCompatActivity
+        activity?.supportActionBar?.setDisplayHomeAsUpEnabled(true)
+        activity?.supportActionBar?.title = "Settings"
+        setHasOptionsMenu(true)
+
+        //Set Email
+        email = arguments?.getString(EMAIL)
+        preferenceScreen.findPreference(resources.getString(R.string.key_profile)).summary = email
+
+        //Set Build Version
+        preferenceScreen.findPreference(resources.getString(R.string.key_version)).title = "Version " + BuildConfig.VERSION_NAME
+    }
+
+    override fun onPreferenceTreeClick(preference: Preference?): Boolean {
+        if (preference?.key == resources.getString(R.string.key_rating)) {
+            //Opens our app in play store
+            startAppPlayStore(activity?.packageName.nullToEmpty())
+            return true
+        }
+        if (preference?.key == resources.getString(R.string.key_suggestion)) {
+            //Links to suggestion form
+            context?.let {
+                Utils.openUrl(it, FORM_LINK)
+            }
+            return true
+        }
+        if (preference?.key == resources.getString(R.string.key_profile)) {
+            //Logout Dialog shown
+            showDialog()
+            return true
+        }
+
+        return false
+    }
+
+    private fun startAppPlayStore(packageName: String) {
+        try {
+            startActivity(Intent(Intent.ACTION_VIEW, Uri.parse(settingsViewModel.getMarketAppLink(packageName))))
+        } catch (error: ActivityNotFoundException) {
+            startActivity(Intent(Intent.ACTION_VIEW, Uri.parse(settingsViewModel.getMarketWebLink(packageName))))
+        }
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        return when (item.itemId) {
+            android.R.id.home -> {
+                activity?.onBackPressed()
+                true
+            }
+            else -> super.onOptionsItemSelected(item)
+        }
+    }
+
+    override fun onDestroyView() {
+        val activity =  activity as? AppCompatActivity
+        activity?.supportActionBar?.setDisplayHomeAsUpEnabled(false)
+        activity?.supportActionBar?.title = "Profile"
+        setHasOptionsMenu(false)
+        super.onDestroyView()
+    }
+
+    private fun showDialog() {
+        val builder = AlertDialog.Builder(activity)
+        builder.setMessage(resources.getString(R.string.message))
+                .setPositiveButton(resources.getString(R.string.logout)) { _, _ ->
+                    if (settingsViewModel.isLoggedIn()) {
+                        settingsViewModel.logout()
+                        startActivity(Intent(context, MainActivity::class.java))
+                        activity?.finish()
+                    }
+                }
+                .setNegativeButton(resources.getString(R.string.cancel)) { dialog, _ -> dialog.cancel() }
+        val alert = builder.create()
+        alert.show()
+    }
+}
```

Comment:
Not a good idea to copy whole fragment, just encapsulate the changes in one component and add two versions of that component